### ISSUE TITLE
feature: check if experimental.affinity is set

### ISSUE THREAD

**Issue by theol0403** [vscode-neovim/vscode-neovim#1267]
https://github.com/vscode-neovim/vscode-neovim/issues/1051

**Comment by theol0403** [comment]
@prabin-lc thanks for looking into this! Yeah, I believe reading the settings is the only option, and I think there are a few other extensions that do this. Look into seeing if there is an official settings API, it does seem like trying to hardcode the filesystem path would be a hacky solution.

**Comment by prabin-lc** [comment]
I looked a bit more into this. Reading the files might be the only option right now.
The settings API (`vscode.workspace.getConfiguration()`) only provides the settings specific to the extension. As the experimental affinity setting is under Feature settings of vscode itself, its not available for individual extension.

The extension host could have been directly checked to see if the extension is running in a non default runtime but unfortunately the host object is private and can't be accessed programtically.
![image](https://github.com/vscode-neovim/vscode-neovim/assets/51957981/e04409a0-ad79-4035-9bbb-007a85dcf7a5)

**Comment by theol0403** [comment]
@prabin-lc nice research.

Another option might be to simply force the option, instead of reading it? Maybe see how "settings cycler" and related extensions are able to write arbitrary keys to the config.

**Comment by prabin-lc** [comment]
Hey @theol0403 ,
Thank you for pointing me to settings cycler. As used there, there is a way to access and update all the settings using the vscode api. The docs was unclear regarding this and I couldn't deduce it before.
I have added similar implementation ,as in settings cycler, to read, confirm and update the affinity value.
I have also added a prompt to restart the extension host after changing the setting.
